xarray/plot/plot.py

```diff
@@ -449,10 +449,28 @@ def newplotfunc(darray, x=None, y=None, figsize=None, size=None,
         if imshow_rgb:
             # Don't add a colorbar when showing an image with explicit colors
             add_colorbar = False
+            # Calculate vmin and vmax automatically for `robust=True`
```

Comment:
Can we move this logic to a separate helper function, e.g., ` _rescale_imshow_rgb(darray, vmin, vmax, robust)`? This method is already getting pretty long.